File: 1903-largest-odd-number-in-string/1903-largest-odd-number-in-string.java
class Solution {
    public String largestOddNumber(String num) {
        StringBuilder sb = new StringBuilder();
        int n = num.length();
        int count=0;
        for(int i=n-1 ; i>=0 ; i--){
            if((num.charAt(i)-'0')%2!=0){
                break;
            }else{
                count++;
            }
        }
        for(int i=0 ; i < n-count ; i++){
            sb.append(num.charAt(i));
        }
        return sb.toString();
    }
}

File: 1903-largest-odd-number-in-string/1903-largest-odd-number-in-string.cs
public class Solution {
    public string LargestOddNumber(string num) {
        int ind=-1;
        StringBuilder sb = new StringBuilder();
        for(int i=num.Length-1 ; i>=0 ;i--){
            if(num[i]%2!=0){
                ind = i;
                break;
            }
        }
        for(int i=0 ; i<=ind ; i++){
            sb.Append(num[i]);
        }
        return sb.ToString();
    }
}